effekt/shared/src/main/scala/effekt/Parser.scala

```diff
@@ -745,6 +746,14 @@ class EffektLexers(positions: Positions) extends Parsers(positions) {
   def someSep[T](p: => Parser[T], sep: => Parser[_]): Parser[List[T]] =
     rep1sep(p, sep) ^^ { _.toList }
 
+  extension [T] (p: Parser[T]) {
+    def !!(errorMessage: T => String): Parser[Nothing] =
+      p.flatMap(t => error(errorMessage(t)))
+
+    def !!!(errorMessage: String): Parser[Nothing] =
+      p.flatMap(_ => error(errorMessage))
+  }
+
```

Comment:
Thanks for these utility functions, they will be really helpful for improving error messages from the parser in the future! :)